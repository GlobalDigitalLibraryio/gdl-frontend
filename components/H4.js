// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import styled from 'styled-components';
import media from './helpers/media';

const H4 = styled.h4`
  font-weight: 500;
  font-size: 16px;
  line-height: 22px;
  ${media.tablet`
    font-size: 20px;
    line-height: 26px;
  `};
  margin: 0;
`;

export default H4;
