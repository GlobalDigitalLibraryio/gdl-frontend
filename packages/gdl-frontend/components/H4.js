// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import styled from 'react-emotion';
import media from '../style/media';
import { fonts } from '../style/theme';

const H4 = styled.h4`
  font-weight: ${fonts.weight.medium};
  font-size: 16px;
  line-height: 22px;
  ${media.tablet`
    font-size: 20px;
    line-height: 26px;
  `};
  margin: 0;
`;

export default H4;
