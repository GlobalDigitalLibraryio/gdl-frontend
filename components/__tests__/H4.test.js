// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';
import H4 from '../H4';

expect.addSnapshotSerializer(createSerializer(emotion));

test('Renders', () => {
  const tree = shallow(<H4>Header 4</H4>);

  expect(toJson(tree)).toMatchSnapshot();
});
