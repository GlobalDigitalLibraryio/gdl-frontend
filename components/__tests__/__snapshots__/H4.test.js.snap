// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.c0 {
  font-weight: 500;
  font-size: 16px;
  line-height: 22px;
}

.c0:first-child {
  margin-top: 0;
}

@media (min-width:769px) {
  .c0 {
    font-size: 20px;
    line-height: 26px;
  }
}

<h4
  className="c0"
>
  Header 4
</h4>
`;
